pages/c/[communityName].js

```diff
@@ -0,0 +1,48 @@
+import Constant from "utils/constant";
+import RedirectUtils from "utils/redirect"
+import UserAgentUtils from "utils/userAgent"
+import parser from 'ua-parser-js';
+import { useEffect } from "react"
+import { useRouter } from "next/router"
+
+export async function getServerSideProps(context) {
+    const { communityName } = context.query
+    let isDynamicLink = communityName?.indexOf('+') >= 0
+    let userAgent = parser(context?.req?.headers['user-agent'])
+    
+    if (UserAgentUtils.isMobile(userAgent)) {
+        let redirect = await RedirectUtils.redirectCommunityForMobileDevice(userAgent, communityName)
+        if (redirect && !isDynamicLink) return redirect
+    }
+
+    if(UserAgentUtils.isIos(userAgent)) {
+        return {
+            redirect: {
+                destination: Constant.Link.appstore,
+            }
+        }
+    }
+
+    if(UserAgentUtils.isAndroid(userAgent)) {
+        return {
+            redirect: {
+                destination: Constant.Link.playstore,
+            }
+        }
+    }
+
+    return {
+        redirect: {
+            destination: Constant.Link.bettersocial,
+        }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getServerSideProps` function has multiple return statements which makes the code harder to read and maintain. It would be better to have a single return statement at the end of the function. You can create a variable `redirectDestination` and assign the appropriate value based on the conditions, then return that variable at the end.

```diff
-    if(UserAgentUtils.isIos(userAgent)) {
-        return {
-            redirect: {
-                destination: Constant.Link.appstore,
-            }
-        }
-    }
-
-    if(UserAgentUtils.isAndroid(userAgent)) {
-        return {
-            redirect: {
-                destination: Constant.Link.playstore,
-            }
-        }
-    }
-
-    return {
-        redirect: {
-            destination: Constant.Link.bettersocial,
-        }
-    }

+    let redirectDestination = Constant.Link.bettersocial;
+
+    if(UserAgentUtils.isIos(userAgent)) {
+        redirectDestination = Constant.Link.appstore;
+    } else if(UserAgentUtils.isAndroid(userAgent)) {
+        redirectDestination = Constant.Link.playstore;
+    }
+
+    return {
+        redirect: {
+            destination: redirectDestination,
+        }
+    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->